---
title: "The Data"
author: "NCBI Hackathon: Proteomic_Correlation_Shiny group"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_document
---
<!--
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{Testing correlation heatmaps}
  %\VignetteEncoding{UTF-8}
-->

## Data Description

#### The DepLabData package uses data from experiments done by a proteomics lab at Weill Cornell.  [Maxquant](http://www.nature.com/nprot/journal/v11/n12/abs/nprot.2016.136.html) outputs 276 variables for 5,208 proteins identified in the PCP experiments (2 replicates each for the EV, or empty vector; WT, or overexpressed ERK2 for recapitulating the epithelial phenotype; and DN, or the mutant ERK2 which accelerates the epithelial to mesenchymal transition.

## Data Cleaning

#### We provided the cleaned data in the rdata objects in /data. Here we outline how the raw data is processed with the function clean_MQ

#### 1. Initial check: The function will stop if no cleaning steps are given.

#### 2. By default, contaminants from the experiment are removed. In proteomics experiments, pig trypsin is used as a control for assessment of the mass spec experiment efficiency. Also, decoys, which is simulated fake peptides generated by MaxQuant and used for calculating false predictions from the mass spectrometer, are removed. 

#### 3. The organism name given to *poi*, is used for extracting proteins with identifiers specific to that organism and no other organism.

#### 4. Spike ins, which by default are none, are removed from the data set. 

```{r,echo=F,eval=F}
#not to be run-only to understand the cleaning function
remove.contaminants = TRUE
remove.decoys = TRUE
poi = NULL
spikeIn = NULL

  # Currently, this function will subset the data.frame more and more, thus
  # multiple filtering options may clash. E.g., if the data.frame is already
  # filtered to only contain trypsin-related entries, it will most likely not
  # find anything related to a Uniprot search for non-trypsin proteins.

  if(dim(mq.df)[1] == 0)(warning("The input to cleaning_MQ is empty."))
  
 mq.out <- mq.df

 if(!remove.contaminants && !remove.decoys && is.null(poi) && is.null(spikeIn)){
    warning("Note that none of the offered filtering options is set.
            The in-going data frame should be the same as the out-going one.")
    }
  
 if(remove.contaminants){
    mq.out <- subset(mq.out, !grepl("CON", mq.out$Protein.IDs))
  }
  
 if(remove.decoys){
    mq.out <- subset(mq.out, !grepl("REV", mq.out$Protein.IDs))
  }
  
 if(!is.null(poi)){
    
   if(poi == "yeast"){
      
     mq.out <- subset(mq.out, grepl("^[YQ]+", Protein.IDs))
      
   }else if(poi == "human"){
      
     # the massive regex in the middle is from TrEMBL (http://www.uniprot.org/help/accession_numbers)
      mq.out <- subset(mq.out, grepl("([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z][0-9]([A-Z][A-Z0-9]{2}[0-9]){1,2})", Protein.IDs))
      
     }else{stop("If you would like to retrieve the proteins of interest for the
                 organism for which the experiment was done, specify one of the
                 available options: ‘yeast’ or ‘human’.")
        }
  }
  
 # extracting spiked-in proteins
  if(!is.null(spikeIn)){
    
   if(!is.null(poi)){
      stop("The option to retrieve spike-in entries is not compatible with
           retrieving yeast or human entries. Set `poi = NULL`.")
    }
    
   if(remove.contaminants){
      warning("Extracting the results for spike-ins while at the same time
              removing contaminants will probably not yield the desired results
              (if anything). Recommended settings: remove.contaminants = FALSE,
              remove.decoys = TRUE, poi = NULL")
    }
    
   if(all(spikeIn == "trypsin")){
      
     mq.out <- subset(mq.out, grepl("P00761$|P00761[^a-zA-Z0-9]", Protein.IDs) &
                           grepl("CON", Protein.IDs))
      }else{
        ID.check <- check_nomenclature(spikeIn)
        if(!all(ID.check)){stop("The ID(s) you supplied to `spikeIn =` do(es)
            not meet the UniProt or yeast gene nomenclature criteria.")}
        reg.1 <- paste(paste(spikeIn, "$", sep = ""), collapse="|")
        reg.2 <- paste(paste(spikeIn, "[^a-zA-Z0-9]", sep = ""), collapse="|")
        reg.combi <- paste(reg.1, reg.2, sep = "|", collapse = "")
        mq.out <- subset(mq.out, grepl( reg.combi, Protein.IDs))
        }
    }
  
 # done cleaning
  if(dim(mq.out)[1] == 0){
    warning("None of the entries in the MaxQuant output survived the cleaning.
            Check that you selected the correct organism for the data that you uploaded.")
    }
  
 dim(mq.out)


#code for cleaning data

#code for reading:
reading_MQ <- function(filename){
  
 mq.in <- read.table(filename, header=TRUE, sep="\t",
                   strip.white = TRUE, fill = TRUE, stringsAsFactors = FALSE,
                   comment.char = "") # important to also capture cases with a # in the Fasta header
  
dim(mq.in)
}

```

#### A short description of the processed datasets are given by pulling up the help pages of the cleaned objects, For example,

```{r,echo=T,eval=T}

path = "/Users/nickgiangreco/GitHub/DepLabData/data/"

objfile<-list.files(path=path)[1]

load( paste0( path, objfile) )

dim(DN_trial1)

require(devtools)
#DepLabData package location in local
#install( "/Users/nickgiangreco/GitHub/DepLabData" )
library(DepLabData)

help(DN_trial1,help_type = "text" )

```

## 2. Data and Variable Description

#### Raw MS data processed by MaxQuant is described by 7 variables. These variables along with definitions and relevant information are described below:
##
##
## MQ Output Variables:

## 1) Raw Intensity
## 2) LFQ (Label-free quantification)
## 3) MS Count
## 4) Peptides count
## 5) Unique Peptides Only
## 6) Razor and Unique Peptides
## 7) Sequence Coverage

#### Definitions:

##### 1) Raw Intensity

##### 2) LFQ (Label-free quantification)

##### 3) MS Count

##### 4) Peptides count

##### 5) Unique Peptides only

##### 6) Razor and Unique Peptides

##### 7) Sequence Coverage This variable refers to the percentage of total amino acids in a 
## known protein that are present in the form of peptides identified in a particular MS experiment
## for which that known protein has been determined to be present. 100% coverage would denote that there
## are peptides present representing all amino acids in the known protein, while 50% would mean half of
## the knonw protein is covered, etc.


## Other:

## Notes on Contaminants
## Notes on Spike-ins
## Notes on Decoy peptides

## Contaminants : Common contaminants include keratin (from human hair, nails, hair) and proteins present in BSA (bovine serum albumin). Peptides identified from species other than the species that is the target of the experiment in question
## indicate outside sources of contamination (for a human experiment, bacterial or mouse proteins may be some common contaminants) s




